pkg/workloadapi/workload.proto

```diff
@@ -17,11 +17,16 @@ syntax = "proto3";
 package istio.workload;
 option go_package="pkg/workloadapi";
 
-// Address represents a unique address, keyed by IP address and network.
-// An Address can represent a Service or an individual workload.
-// Address is intended to be able to be looked up on-demand, allowing a workload
+// Address represents a unique address.
+//
+// Address joins two sub-resources, Workload and Service, to support querying by IP address.
+// Address is intended to be able to be looked up on-demand, allowing a client
 // to answer a question like "what is this IP address", similar to a reverse DNS lookup.
-// Keyed by network/IP address. Example: "default/1.2.3.4".
+//
+// Each resource will have a mesh-wide unique opaque name, defined in the individual messages.
```

Comment:
This seems like another NIH istio-reinvented network concept and syntax and doesn't make a clear or good protocol IMO. 

Mesh wide name is fine - not sure why 'opaque' when it is defined bellow in the network/IP syntax.
I think the FQDN is the correct 'mesh wide name'. Not sure what 'unique' means, each resource has many mesh-wide names it can be referenced to. 

I haven't seen the doc and details on how we want to handle multi-network - but ideally it would not leak into the resource names. It also doesn't make too much sense from behavior perspective - ztunnel captures by IP, so the only network it is queriying for is its own IP ( which should be in the node meta, not appended to each IP). I don't see any case where a ztunnel will even be exposed to network names - it may get back some east-west gateways in the response, but it won't be able to do much with the info about the network.

I would suggest removing the multi-network aspects until a design is more clearly defined and agreed.